File: src/Chapter-6-C99/Questions/Question11.c
//The value of the mathematical constant e can be expressed as an infinite series:
//e = 1 + 1/1! + 1/2! + 1/3! +..
//Write a program that approximates e by computing the value of
//1 + 1/1! + 1/2! + 1/3! + ... + 1/n!
//where n is an integer entered by the user
#include <stdio.h>

int main(void)
{
    int n;
    float e = 1.00f, fact;
    printf("Please enter a value of n: ");
    scanf("%d", &n);
    for (int i = 1; i <= n; i++)
    {
        fact = 1;
        for (int j = 1; j <= i; j++)
            fact *= j;
        e = e + (1/fact);
    }
    printf("Value of e is %.2f", e);
    return 0;
}

File: src/Chapter-6-Rust/Questions/Question11.rs
//The value of the mathematical constant e can be expressed as an infinite series:
//e = 1 + 1/1! + 1/2! + 1/3! +..
//Write a program that approximates e by computing the value of
//1 + 1/1! + 1/2! + 1/3! + ... + 1/n!
//where n is an integer entered by the user
#[macro_use] extern crate text_io;
use std::process::Command;
//Need this to use the print!
use std::io::{self, Write};
fn main()
{
	let n:i32;
	let mut e : f32 = 1.0;
	let mut fact : f32;
	print!("Please enter a value of n: ");
	io::stdout().flush().unwrap();
	scan!("{}", n);
	for i in 2..n
	{
		fact = 1.00;
		for j in 1..i
		{
			fact *= j as f32;
		}
		e = e + (1.0/fact);
	}
	print!("Value of e is {:.5}\n", e);
	io::stdout().flush().unwrap();
	let _ = Command::new("cmd.exe").arg("/c").arg("pause").status();
}
